### PR TITLE
Controlled button styles

### PR DESCRIPTION
**Pull request by dprzybylek**
new controlled (and group) button styles:
group button

### CODE REVIEW

File: packages/design-system/src/components/ButtonGroup/ButtonGroup.js [hunk 1]
```diff
@@ -8,26 +8,26 @@ const baseClass = 'btn-group';
 
 class ButtonGroup extends React.Component {
   state = {
-    currentIndex: defaultIndex
+    currentindex: defaultIndex
```
Comment (marcinsawicki) [inline packages/design-system/src/components/ButtonGroup/ButtonGroup.js]: Why you removed the camelcase format?

File: packages/design-system/src/components/ButtonGroup/ButtonGroup.md [hunk 2]
```diff
@@ -22,22 +22,29 @@ By default ButtonGroup use internal state to track currently selected button. It
...
 
-If you want to use ButtonGroups as controlled component you can set `currentIndex` props to set which button is active.
+If you want to use ButtonGroups as controlled component you can set `currentindex` props to set which button is active.
 
 ```js
-<ButtonGroup currentIndex={1}>
+initialState = { currentindex: 0 };
+
+handleClick = (index, e) => {
+  console.log('index', index)
```
Comment (walaszczykm) [inline packages/design-system/src/components/ButtonGroup/ButtonGroup.md]: this `console.log` probably should not be here

File: packages/design-system/src/components/ButtonGroup/ButtonGroup.js (hunk 1 above)
Comment (walaszczykm) [reply packages/design-system/src/components/ButtonGroup/ButtonGroup.js]: Yes I also opt for the original one that I can't find any reasoning for not making it in camel case as they are two separate words 🤔

File: packages/design-system/src/components/ButtonGroup/ButtonGroup.js [hunk 3]
```diff
@@ -58,7 +58,7 @@ class ButtonGroup extends React.Component {
 }
 
 ButtonGroup.defaultProps = {
-  onIndexChange: () => {}
+  onIndexChange: (index, event) => {}
```
Comment (walaszczykm) [inline packages/design-system/src/components/ButtonGroup/ButtonGroup.js]: It is actually the same: still `noop` function but now you have there not used arguments which could be bad in case of some linting rule in future.

File: packages/design-system/src/components/ButtonGroup/ButtonGroup.js (hunk 1 above)
Comment (dprzybylek) [reply packages/design-system/src/components/ButtonGroup/ButtonGroup.js]: `const {... currentIndex, ...restProps } = this.props;` this fixed my issue, reverted curentIndex at b1005187a0d51f7e88f2f764c0701d21ca564965

File: packages/design-system/src/components/ButtonGroup/ButtonGroup.md (hunk 2 above)
Comment (dprzybylek) [reply packages/design-system/src/components/ButtonGroup/ButtonGroup.md]: removed at b1005187a0d51f7e88f2f764c0701d21ca564965

File: packages/design-system/src/components/ButtonGroup/ButtonGroup.js (hunk 3 above)
Comment (dprzybylek) [reply packages/design-system/src/components/ButtonGroup/ButtonGroup.js]: deleted this noop, and added check in call b1005187a0d51f7e88f2f764c0701d21ca564965

File: packages/design-system/src/components/ButtonGroup/ButtonGroup.md [hunk 4]
```diff
@@ -25,19 +25,25 @@ By default ButtonGroup use internal state to track currently selected button. It
...
+    borderRadius: '50%',
+    display: 'inline-block',
+    position: 'relative',
+    bottom: '5px'
+  }}></span></Button>
   <Button>Third Button</Button>
 </ButtonGroup>
 ```
-
-```js noeditor
```
Comment (kamilmateusiak) [inline packages/design-system/src/components/ButtonGroup/ButtonGroup.md]: why did you remove html example?

File: packages/design-system/src/components/ButtonGroup/ButtonGroup.md [hunk 5]
```diff
@@ -25,19 +25,25 @@ By default ButtonGroup use internal state to track currently selected button. It
...
+handleClick = (index, e) => {
+  setState({
+    currentIndex: index
+  })
+}
+
+<ButtonGroup currentIndex={state.currentIndex} onIndexChange={handleClick}>
   <Button>First Button</Button>
-  <Button>Second Button</Button>
+  <Button>Second Button <span style={{
```
Comment (kamilmateusiak) [inline packages/design-system/src/components/ButtonGroup/ButtonGroup.md]: could you declare this span as separate component (here, in doc file). It would be more readable that way
Is it element of design-system?

File: packages/design-system/src/components/ButtonGroup/ButtonGroup.js [hunk 6]
```diff
@@ -8,43 +8,43 @@ const baseClass = 'btn-group';
...
+    const mappedChildren = React.Children.map(children, (child, i) =>
       React.cloneElement(child, {
         size,
         fullWidth,
         primary: false,
         secondary: true,
         destructive: false,
         onClick: event => {
-          this.handleClick(index, event);
+          this.handleClick(i, event);
```
Comment (kamilmateusiak) [inline packages/design-system/src/components/ButtonGroup/ButtonGroup.js]: it should have property `type='button'`

File: packages/design-system/src/components/ButtonGroup/ButtonGroup.js [hunk 7]
```diff
@@ -8,43 +8,43 @@ const baseClass = 'btn-group';
...
-    this.props.onIndexChange(index, event);
+    this.props.onIndexChange && this.props.onIndexChange(index, event);
     this.setState({ currentIndex: index });
   };
 
   render() {
-    const { currentIndex } = this.state;
-    const { size, fullWidth, children, className, ...restProps } = this.props;
+    const { index } = this.state;
+    const { size, fullWidth, children, className, onIndexChange, currentIndex, ...restProps } = this.props;
```
Comment (walaszczykm) [inline packages/design-system/src/components/ButtonGroup/ButtonGroup.js]: you do not need `currentIndex` here as it is not used in the component `render` method

Comment (kamilmateusiak) [reply packages/design-system/src/components/ButtonGroup/ButtonGroup.js]: it's not used, but there is `restProps` from the spread operator. Below div will receive currentIndex as property, but there is no property like this on div element.

Comment (dprzybylek) [reply packages/design-system/src/components/ButtonGroup/ButtonGroup.js]: ☝️

File: packages/design-system/src/components/ButtonGroup/ButtonGroup.md (hunk 5 above)
Comment (dprzybylek) [reply packages/design-system/src/components/ButtonGroup/ButtonGroup.md]: no, this was just example for designer, removed

File: packages/design-system/src/components/ButtonGroup/ButtonGroup.js (hunk 6 above)
Comment (kamilmateusiak) [reply packages/design-system/src/components/ButtonGroup/ButtonGroup.js]: what about that?
